java/maven/src/org/netbeans/modules/maven/NbMavenProjectImpl.java

```diff
@@ -374,7 +391,8 @@ public MavenProject loadParentOf(MavenEmbedder embedder, MavenProject project) t
         request.setRepositorySession(maven.newRepositorySession(req));
 
         if (project.getParentFile() != null) {
-            parent = builder.build(project.getParentFile(), request).getProject();
```

Comment:
Former call to `DefaultProjectBuilder.build` does not contain error handling / processing as implemented in `MavenProjectCache`. The other branch works with (nonlocal) artifacts, it is still TBD but serves mainly as a fallback.